test/integration/pool-test.js

```diff
@@ -442,3 +442,43 @@ tap.test(
       .catch(t.threw);
   }
 );
+
+tap.test("pool destroys a resource when maxUses is reached", t => {
+  const resourceFactory = new ResourceFactory();
+
+  const factory = {
+    name: "test20",
+    create: resourceFactory.create.bind(resourceFactory),
+    destroy: resourceFactory.destroy.bind(resourceFactory),
+    validate: resourceFactory.validate.bind(resourceFactory),
+    max: 1,
+    min: 0,
+    maxUses: 2,
+    idleTimeoutMillis: 100,
+    acquireTimeoutMillis: 100
+  };
+
+  const pool = new Pool(factory);
+
+  pool.acquire().then(clientA => {
+    // first use
+    pool.release(clientA);
+    pool.acquire().then(clientB => {
+      // second use
+      t.equal(clientA, clientB);
+
+      // should end-of-life the resource
+      pool.release(clientA);
+
+      pool.acquire().then(clientC => {
+        t.notEqual(clientB, clientC);
+      });
+    });
+  });
+
+  // assert the first connection was destroyed once the max-use limit was reached
+  setTimeout(() => {
+    t.equal(0, resourceFactory.bin[0].id);
+    t.end();
+  }, 100);
```

Comment:
Why is `setTimeout` being used here? Can't this fit into the promise chain above?

Reply:
I simply followed the pattern used in the other tests in the test case that assert the destroy bin.  I'm open to changing it, and the others if we want to be consistent.

My silent assumption is there was a reason the original authors did it this way, but I admit I did not investigate.

Reply:
I see, I suspected that might be the case. If you can please investigate it further it would be great. Don't worry about the others for now, you can keep them unchanged.